File: 376-wiggle-subsequence/376-wiggle-subsequence.cs
public class Solution {
    public int WiggleMaxLength(int[] nums) {
        
        int count = 1;
        int prevNumber;
        int prevDifference = 0;
        int difference;
        
        if(nums.Length == 1) return count;
        
        for(int i = 1; i < nums.Length; i++) {
            prevNumber = nums[i-1];
            if(nums[i] == prevNumber) continue;
            else {
                difference = nums[i] - prevNumber;
                if((difference > 0 && prevDifference <= 0) ||(difference < 0 && prevDifference >= 0)) {
                    count++;
                    prevDifference = difference;
                }
                else continue;
            }
        }
        
        return count;
    }
}

File: 376-wiggle-subsequence/376-wiggle-subsequence.py
class Solution:
    def wiggleMaxLength(self, nums: List[int]) -> int:
        length = len(nums)
        if length <= 1:
            return length
        
        up, down = 1, 1
        
        for i in range(length - 1):
            if nums[i] < nums[i+1]:
                up = down + 1
            elif nums[i] > nums[i+1]:
                down = up + 1
        
        return max(up, down)
            